alf/algorithms/diayn_algorithm.py

```diff
@@ -106,39 +117,40 @@ def train_step(self,
         observations_aug = time_step.observation
         step_type = time_step.step_type
         observation, skill = observations_aug
-        prev_skill = state
+        prev_skill = state.detach()
+
+        # normalize observation for easier prediction
+        if self._observation_normalizer is not None:
+            observation = self._observation_normalizer.normalize(observation)
 
         if self._encoding_net is not None:
             feature, _ = self._encoding_net(observation)
 
-        skill_pred, _ = self._discriminator_net(inputs=feature)
+        skill_pred, _ = self._discriminator_net(feature)
 
-        skill_discriminate_loss = tf.nn.softmax_cross_entropy_with_logits(
-            labels=prev_skill, logits=skill_pred)
+        if self._skill_spec.is_discrete:
+            loss = torch.nn.CrossEntropyLoss(reduction='none')(
+                input=skill_pred, target=prev_skill.to(torch.int64))
+        else:
+            # nn.MSELoss doesn't support reducing along a dim
+            loss = torch.sum(math_ops.square(skill_pred - prev_skill), dim=-1)
```

Comment:
Might be better to use ```mean``` for reducing. ```sum``` is more easily impacted by the dims of the skill vectors.

Reply:
It's nothing wrong with sum because it's essentially the log likelihood of a normal distribution with a fixed std.